### PR TITLE
PHPStan Fixes

### PR DESCRIPTION
**Pull request by MGatner**
**Description**
* Adds **composer.json** to the path list for PHPStan Action
* Specifies the Same Site Cookie string values for PHPStan compliance
* Removes the ignore line for a now-passing check

See https://github.com/codeigniter4/CodeIgniter4/pull/4134

**Checklist:**
- [X] Securely signed commits
- [X] Component(s) with PHPdocs
- [X] Unit testing, with >80% coverage
- n/a User guide updated
- [X] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
It appears that Rector does not understand explicit strings for PHPDoc types, which PHPStan requires. @samsonasik or @paulbalandan have any suggestions?

**Comment by MGatner** [comment]
E.g.
```
2) app/Config/Security.php
28

29
    ---------- begin diff ----------
30
--- Original
31
+++ New
32
@@ -86,7 +86,7 @@
33
 	 * Defaults to `Lax` as recommended in this link:
34
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
35
 	 *
36
-	 * @var 'Lax'|'None'|'Strict'
37
+	 * @var Lax|None|Strict
38
 	 */
39
 	public $samesite = 'Lax';
40
 }
41
    ----------- end diff -----------
```

**Comment by samsonasik** [comment]
That seems related to Format preserving doc, ref :

- https://github.com/nikic/PHP-Parser/issues/344
- https://github.com/rectorphp/rector/issues/4334

I may create a workaround patch for that in rector side, eg in the ContentPatcher class:

https://github.com/rectorphp/rector/blob/133322cc658fcd8d5ba162b4a18c7d9d88eba714/src/PhpParser/Printer/ContentPatcher.php#L142-L147

For now, I think we can do with:

```php
/** @var string 'Lax'|'None'|'Strict' */
```

ref :
- https://getrector.org/demo/5aea2ecd-f2db-4f60-b61d-ccfa06d9c457 
- https://phpstan.org/r/03e63c8d-6d30-4bed-bb76-1ca36a0b679c

**Comment by paulbalandan** [comment]
Try this if you need to be explicit of the type in phpstan.

```php
// for vars
 * @var string
 * @phpstan-var 'Lax'|'None'|'Strict'

// for returns
 * @return object
 * @phpstan-return TObject
```

Seems phpstan understand those pseudo-docblocks just like in psalm.

**Comment by samsonasik** [comment]
@paulbalandan the `@phpstan-var 'Lax'|'None'|'Strict'` is  not work for rector post -processing right now ref https://getrector.org/demo/1d208398-63da-4f18-8e44-00e16b72bb3e

Need a workaround patch in rector side while wait for a real fix in nikic/php-parser side.

**Comment by paulbalandan** [comment]
It actually works for me though. You need to have both `@var` and `@phpstan-var`.

https://getrector.org/demo/7ed7b50f-7193-4f45-bade-322d372d7343

**Comment by samsonasik** [comment]
Ok, that seems a solution, but need different `/* */` block to make work in phpstan ref 

https://phpstan.org/r/15ddc4ad-57a1-4d3b-ac04-bbb2075dda73

vs

https://phpstan.org/r/421f43cf-2eac-4f5f-98c2-929fbca30543

**Comment by paulbalandan** [comment]
No need. You just have a typo in your second phpstan ref

**Comment by samsonasik** [comment]
@paulbalandan oh yeah, you're correct 👍

**Comment by MGatner** [comment]
Thanks for taking a look guys. Can someone either push the final version you agreed on or suggest it in a review?

File: app/Config/App.php [hunk 1]
```diff
@@ -303,7 +303,7 @@ class App extends BaseConfig
 	 * (empty string) means no SameSite attribute will be set on cookies. If
 	 * set to `None`, `$cookieSecure` must also be set.
 	 *
-	 * @var string
+	 * @var 'Lax'|'None'|'Strict'
```
Comment (paulbalandan) [inline app/Config/App.php:306]:
```suggestion
	 * @var string
	 * @phpstan-var 'Lax'|'None'|'Strict'
```

File: app/Config/Security.php [hunk 2]
```diff
@@ -86,7 +86,7 @@ class Security extends BaseConfig
 	 * Defaults to `Lax` as recommended in this link:
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
 	 *
-	 * @var string
+	 * @var 'Lax'|'None'|'Strict'
```
Comment (paulbalandan) [inline app/Config/Security.php:89]:
```suggestion
	 * @var string
	 * @phpstan-var 'Lax'|'None'|'Strict'
```

File: system/Security/Security.php [hunk 3]
```diff
@@ -99,7 +99,7 @@ class Security implements SecurityInterface
 	 * Defaults to `Lax` as recommended in this link:
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
 	 *
-	 * @var string
+	 * @var 'Lax'|'None'|'Strict'
```
Comment (paulbalandan) [inline system/Security/Security.php:102]:
```suggestion
	 * @var string
	 * @phpstan-var 'Lax'|'None'|'Strict'
```

File: system/Session/Session.php [hunk 4]
```diff
@@ -125,7 +125,7 @@ class Session implements SessionInterface
 	 * Cookie SameSite setting as described in RFC6265
 	 * Must be 'None', 'Lax' or 'Strict'.
 	 *
-	 * @var string
+	 * @var 'Lax'|'None'|'Strict'
```
Comment (paulbalandan) [inline system/Session/Session.php:128]:
```suggestion
	 * @var string
	 * @phpstan-var 'Lax'|'None'|'Strict'
```

File: app/Config/App.php (hunk 1 above)
Comment (totoprayogo1916) [reply app/Config/App.php:306]: and for empty string?

File: app/Config/Security.php (hunk 2 above)
Comment (samsonasik) [inline app/Config/Security.php:89]:
```suggestion
       * @var string
       * @phpstan-var 'Lax'|'None'|'Strict'
 ```

File: system/Security/Security.php (hunk 3 above)
Comment (samsonasik) [inline system/Security/Security.php:102]:
```suggestion
       * @var string
       * @phpstan-var 'Lax'|'None'|'Strict'
 ```

File: system/Session/Session.php [hunk 5]
```diff
@@ -125,7 +125,7 @@ class Session implements SessionInterface
 	 * Cookie SameSite setting as described in RFC6265
 	 * Must be 'None', 'Lax' or 'Strict'.
```
Comment (samsonasik) [inline system/Session/Session.php:126]:
```suggestion
       * @var string
       * @phpstan-var 'Lax'|'None'|'Strict'
 ```

File: app/Config/App.php (hunk 1 above)
Comment (paulbalandan) [reply app/Config/App.php:306]: empty string is really not allowed either in the cookie spec or in the php code. we only use it to define that _no samesite_ will be set. But this behavior is now "obsolete" as MDN reports that modern browsers requires cookies to have `SameSite=Lax` if no samesite is given. This error is thrown in the console.

**Comment by MGatner** [comment]
Thank you both. GitHub is not allowing me to commit revisions for some reason, looks like a bug, but I will get these applied later when I can get back to my server with this branch.

**Comment by totoprayogo1916** [comment]
Is it possible to use this : `@var string[Lax|None|Strict]` ?

**Comment by samsonasik** [comment]
@MGatner @paulbalandan the `@phpstan-var 'Lax'|'None'|'Strict'` seems only working in rector 0.9. I think for now, we can use:

```php
/** @var string 'Lax'|'None'|'Strict' */
```

as I previously suggested. 

@totoprayogo1916 this patch want a quoted doc.

File: app/Config/App.php [hunk 6]
```diff
@@ -304,6 +304,7 @@ class App extends BaseConfig
 	 * set to `None`, `$cookieSecure` must also be set.
 	 *
 	 * @var string
+	 * @phpstan-var 'Lax'|'None'|'Strict'
```
Comment (samsonasik) [inline app/Config/App.php:307]:
```suggestion
       * @var string 'Lax'|'None'|'Strict'
```

File: app/Config/Security.php [hunk 7]
```diff
@@ -87,6 +87,7 @@ class Security extends BaseConfig
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
 	 *
 	 * @var string
+	 * @phpstan-var 'Lax'|'None'|'Strict'
```
Comment (samsonasik) [inline app/Config/Security.php:90]:
```suggestion
       * @var string 'Lax'|'None'|'Strict'
```

File: system/Security/Security.php [hunk 8]
```diff
@@ -100,6 +100,7 @@ class Security implements SecurityInterface
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
 	 *
 	 * @var string
+	 * @phpstan-var 'Lax'|'None'|'Strict'
```
Comment (samsonasik) [inline system/Security/Security.php:103]:
```suggestion
       * @var string 'Lax'|'None'|'Strict'
```

File: system/Session/Session.php [hunk 9]
```diff
@@ -126,6 +126,7 @@ class Session implements SessionInterface
...
 	 *
 	 * @var string
+	 * @phpstan-var 'Lax'|'None'|'Strict'
```
Comment (samsonasik) [inline system/Session/Session.php:129]:
```suggestion
       * @var string 'Lax'|'None'|'Strict'
```

File: app/Config/App.php [hunk 10]
```diff
@@ -304,6 +304,7 @@ class App extends BaseConfig
 	 * set to `None`, `$cookieSecure` must also be set.
 	 *
 	 * @var string
```
Comment (samsonasik) [inline app/Config/App.php:306]:
```suggestion
```

File: app/Config/Security.php [hunk 11]
```diff
@@ -87,6 +87,7 @@ class Security extends BaseConfig
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
 	 *
 	 * @var string
```
Comment (samsonasik) [inline app/Config/Security.php:89]:
```suggestion
```

File: system/Security/Security.php [hunk 12]
```diff
@@ -100,6 +100,7 @@ class Security implements SecurityInterface
 	 * @see https://portswigger.net/web-security/csrf/samesite-cookies
 	 *
 	 * @var string
```
Comment (samsonasik) [inline system/Security/Security.php:102]:
```suggestion
```

File: system/Session/Session.php [hunk 13]
```diff
@@ -126,6 +126,7 @@ class Session implements SessionInterface
 	 * Must be 'None', 'Lax' or 'Strict'.
 	 *
 	 * @var string
```
Comment (samsonasik) [inline system/Session/Session.php:128]:
```suggestion
```

**Comment by samsonasik** [comment]
I am going to fix it right now, I will try to fix the error affected in system/Security/Security.php and system/Session/Session.php    in separate PR

**Comment by samsonasik** [comment]
Thank you all, I created new PR for temporary ignore the phpstan notice in Security and Session class, with `@todo` to clean up in 4.1 for rector 0.9 https://github.com/codeigniter4/CodeIgniter4/pull/4146

**Comment by MGatner** [comment]
Thanks all. Sorry to be MIA finishing this up - special thanks to @samsonasik

**Pull request merged**